File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
input = sys.stdin.readline
def dfs(n):
    visitied[n] = 1
    for v in graph[n]:
        if visitied[v] == 0:
            dfs(v)
N = int(input())
num_edge = int(input())
visitied = [0]*N
graph = [[] for i in range(N)]
for _ in range(num_edge):
    a,b = map(int,input().split())
    graph[a-1].append(b-1)
    graph[b-1].append(a-1)
dfs(0)
print(sum(visitied[1:]))

File: 백준/Silver/2606. 바이러스/바이러스.java

import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;
import java.util.stream.IntStream;
public class Main{
    static boolean[] visited;
    static List<List<Integer>> graph;

    private static void dfs(int n){
        visited[n] = true;
        for(int v : graph.get(n)){
            // 방문하지 않은 노드
            if(visited[v] == false) {
                dfs(v);
            }
        }

    }
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        //BufferedReader br = new BufferedReader(new FileReader("input.txt"));
        
        int N = Integer.parseInt(br.readLine());
        int num_edge = Integer.parseInt(br.readLine());

        visited = new boolean[N];
        graph = new ArrayList<>();
        for(int i=0; i<N; i++){
            graph.add(new ArrayList<>());
        }

        for(int i = 0; i<num_edge; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            graph.get(a-1).add(b-1);
            graph.get(b-1).add(a-1);
        }
        dfs(0);
        
        // boolean[]를 Boolean[]로 변환한 후 스트림을 사용
        long count = IntStream.range(1,N)
                            .filter(i -> visited[i])  // true인 값만 필터링
                            .count();  // 개수 세기

        System.out.println(count);  // 출력

    }

    
}